.github/workflows/push.yml

```diff
@@ -42,6 +42,53 @@ env:
   CC: ccache gcc
   CXX: ccache g++
 jobs:
+  CODE_FORMATTING_CHECK:
+    name: CODE_FORMATTING_CHECK
+    if: github.repository_owner == 'php' || github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: apt
+        run: |
+          set -x
+          sudo apt-get update
+          sudo apt-get install \
+            git \
+            clang-format \
+            php-cli \
+            netcat-openbsd
+      - name: check code formatting
+        run: |
+          set -x
+          git fetch origin master
+          git diff origin/master > ../master.diff
```

Comment:
you should be probably using three-dot Git diff comparison

Reply:
humans don't see this diff, the diff humans see is the last `git diff` in format_pull_request.php, this git just need to be `git apply`-compatible. Do you think it will be easier to debug if something goes wrong, with a three-dot diff here?